### PR TITLE
sandbox: better log output

### PR DESCRIPTION
**Pull request by xu-cheng**
* use syslog filter instead of grep.
* output sandbox log to stdout when verbose and failed.
* output nothing if sandbox log is empty.

cc @mikemcquaid

### CODE REVIEW

File: Library/Homebrew/sandbox.rb [hunk 1]
```diff
@@ -91,24 +98,36 @@ def exec(*args)
...
+      unless logs.empty?
+        @log.write Sandbox.log_warning_message
+        @log.write logs
+
+        if @failed && ARGV.verbose?
+          ohai "Sandbox log:"
+          puts Sandbox.log_warning_message
+          puts logs
+          ohai "Sandbox profile:"
+          puts @profile.dump
```
Comment (mikemcquaid) [inline Library/Homebrew/sandbox.rb]: Do you find this message useful? If so we can keep it otherwise let's :skull:

File: Library/Homebrew/sandbox.rb [hunk 2]
```diff
@@ -91,24 +98,36 @@ def exec(*args)
...
         -o
         -k Time ge #{@start.to_i}
+        -k Message S deny
         -k Sender sandboxd
       ]
-      quiet_system "syslog #{syslog_args * " "} | grep deny > #{@log}"
+      logs = Utils.popen_read("syslog", *syslog_args)
+      unless logs.empty?
+        @log.write Sandbox.log_warning_message
+        @log.write logs
```
Comment (mikemcquaid) [inline Library/Homebrew/sandbox.rb]: Can we `grep` this for any string or is there no prefix?

File: Library/Homebrew/sandbox.rb (hunk 1 above)
Comment (xu-cheng) [inline Library/Homebrew/sandbox.rb]: It's useful to people who are unfamiliar to homebrew sandbox's rule. Otherwise, it's not as they are almost constant.

File: Library/Homebrew/sandbox.rb (hunk 2 above)
Comment (xu-cheng) [inline Library/Homebrew/sandbox.rb]: You mean the log format? 

Currently it's `$((Time)(local)) $(Sender)[$(PID)]: $(Message)`. e.g. `Aug 27 21:19:32 sandboxd[295]: ruby(93659) deny file-write-create /1`. We can tweak it if you want.

File: Library/Homebrew/sandbox.rb (hunk 1 above)
Comment (mikemcquaid) [inline Library/Homebrew/sandbox.rb]: Yeh, I guess I personally find this output kinda noise and hard to follow. Maybe we could add a few newlines or separators in here to split it up a bit if we leave it in?

Comment (xu-cheng) [inline Library/Homebrew/sandbox.rb]: Or we remove this. And add it to document, say `sandbox.md`?

File: Library/Homebrew/sandbox.rb (hunk 2 above)
Comment (mikemcquaid) [inline Library/Homebrew/sandbox.rb]: :cool: looks good :+1:

File: Library/Homebrew/sandbox.rb (hunk 1 above)
Comment (mikemcquaid) [inline Library/Homebrew/sandbox.rb]: Yeh, maybe just remove it? A document will get outdated compared to the code.

**Comment by xu-cheng** [comment]
Rewrote PR to address the comments.

File: Library/Homebrew/sandbox.rb [hunk 3]
```diff
@@ -32,7 +32,13 @@ def initialize
...
   def record_log(file)
-    @log = file
+    @logfile = file
+  end
+
+  def self.log_warning_message; <<-EOS.undent
+      We use time to filter sandbox log. Therefore, unrelated logs may be recorded in below.
+
+    EOS
   end
```
Comment (mikemcquaid) [inline Library/Homebrew/sandbox.rb]: I think we can probably skip printing this message unless it's causing confusion.

**Comment by mikemcquaid** [comment]
Looks good to me!

File: Library/Homebrew/sandbox.rb (hunk 3 above)
Comment (xu-cheng) [inline Library/Homebrew/sandbox.rb]: How about only writing it to log file? Usually there won't be unrelated logs for bot. But on users computer, the chance is high which can cause confusion.

Comment (mikemcquaid) [inline Library/Homebrew/sandbox.rb]: Yep, if we write it maybe after we print the log then that makes sense to me. Equally if we didn't write it if `HOMEBREW_DEVELOPER` is set or something? Up to you!

Comment (xu-cheng) [inline Library/Homebrew/sandbox.rb]: Finally, I decide to only write this message to the bottom of file.

**Comment by mikemcquaid** [comment]
:+1:
